web3/middleware/signing.py

```diff
@@ -0,0 +1,78 @@
+from eth_keys.datatypes import PrivateKey
+from eth_utils import is_same_address
+
+from web3.utils.datastructures import HexBytes
+from web3.exceptions import InvalidAddress
+# from web3.utils.signing import LocalAccount
+
+import logging
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def construct_transaction_signing_middleware(private_key):
+    """Intercept a transaction in flight and sign it.
+
+    Args:
+        private_key (TYPE): Private key used to sign the transaction.
+                            Accepts the following formats:
+                            - `PrivateKey` object from `eth-keys`
+                            - `LocalAccount` object from `web3.eth.account`
+                            - raw bytes
+                            - TODO: hex encoded string
+
+
+    Returns:
+        TYPE: Description
+    """
+
+    # Check if the private_key is either
+    # - eth_keys.keys.PrivateKey
+    # - web3.eth.account.create().privateKey -- can just check that its HexBytes
+    # - bytes
+    # TODO: - hexbytes
+    valid_key_types = (PrivateKey, HexBytes, bytes)
+
+    _private_key = private_key if isinstance(private_key, valid_key_types) else None
+    if not _private_key:
```

Comment:
These three lines could be replaced by the following which I believe is easier to read.

```
if not isinstance(private_key, valid_key_types):
    raise ...
```